### PR TITLE
Custom template for transtions

### PR DESCRIPTION
**Pull request by ar4s**

### CODE REVIEW

File: docs/development/transitions.md [hunk 1]
```diff
@@ -1,6 +1,16 @@
 # Transitions
 The transition change state (e.g. status) of object from one to another - that's helps in product lifecycle management. For each object (asset, support, licence) you can define some workflow (set of transitions) and provide special actions for each transition.
 
+## Custom template for transition
+For each a transition you can separately specify Django's template in the transition's form, to do this add new item to ``TRANSITION_TEMPLATES`` in your settings file. For instance:
```
Comment (mkurek) [inline docs/development/transitions.md]: For each transition

File: src/ralph/lib/transitions/checks.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+                    ))
+                    continue
+                try:
+                    get_template(path)
+                except TemplateDoesNotExist:
+                    errors.append(Error(
+                        'Template {} ({}) doesn\'t exist'.format(
+                            template, path
+                        ),
+                        hint='Check TEMPLATES settings',
```
Comment (mkurek) [inline src/ralph/lib/transitions/checks.py]: did you mean `TRANSITION_TEMPLATES`?

File: src/ralph/lib/transitions/forms.py [hunk 3]
```diff
@@ -13,6 +13,9 @@
 )
 
 
+TRANSITION_TEMPLATES = getattr(settings, 'TRANSITION_TEMPLATES', None)
```
Comment (mkurek) [inline src/ralph/lib/transitions/forms.py]: why getattr? add it to base settings and use `TRANSITION_TEMPLATES = settings.TRANSITION_TEMPLATES` here

File: src/ralph/lib/transitions/models.py [hunk 4]
```diff
@@ -511,6 +511,7 @@ class Transition(models.Model):
     source = JSONField()
     target = models.CharField(max_length=50)
     actions = models.ManyToManyField('Action')
+    template_name = models.CharField(max_length=255, blank=True, default='')
```
Comment (mkurek) [inline src/ralph/lib/transitions/models.py:514]: what will (should?) happen if I save template_name to DB and then remove it from settings?

**Review by mkurek — CHANGES_REQUESTED** [review]

File: src/ralph/lib/transitions/models.py (hunk 4 above)
Comment (ar4s) [reply src/ralph/lib/transitions/models.py:514]: I will add some check for this which emit some warning.

**Review by mkurek — APPROVED** [review]

**Pull request merged**